components/emoji/emoji_list_item/emoji_list_item.tsx

```diff
@@ -2,35 +2,35 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import PropTypes from 'prop-types';
 
 import Permissions from 'mattermost-redux/constants/permissions';
 import {Client4} from 'mattermost-redux/client';
+import {CustomEmoji} from 'mattermost-redux/types/emojis';
+import {Team} from 'mattermost-redux/src/types/teams';
 
 import DeleteEmoji from 'components/emoji/delete_emoji_modal.jsx';
 import AnyTeamPermissionGate from 'components/permissions_gates/any_team_permission_gate';
 
-export default class EmojiListItem extends React.PureComponent {
-    static propTypes = {
-        emoji: PropTypes.object.isRequired,
-        currentUserId: PropTypes.string.isRequired,
-        creatorDisplayName: PropTypes.string.isRequired,
-        creatorUsername: PropTypes.string,
-        currentTeam: PropTypes.object,
-        onDelete: PropTypes.func,
-        actions: PropTypes.shape({
-            deleteCustomEmoji: PropTypes.func.isRequired,
-        }).isRequired,
+export type Props = {
+    emoji: CustomEmoji;
+    currentUserId: string;
+    creatorDisplayName: string;
+    creatorUsername?: string;
+    currentTeam?: Team;
+    onDelete?: (id: string) => void;
+    actions: {
+        deleteCustomEmoji: (id: string) => void;
```

Comment:
I think this `deleteCustomEmoji` should be like this
```ts
deleteCustomEmoji: (emojiId: string) => Promise<{data: boolean, error: ServerError}>
```
https://github.com/mattermost/mattermost-redux/blob/4aa23838f96f94326bdc8b291d7674079da8bb5d/src/actions/emojis.ts#L186
@cpoile Any thoughts?

Reply:
Good catch @Ayanrocks -- but I think it needs to be:

```suggestion
        deleteCustomEmoji: (id: string) => ActionFunc;
```